src/Microsoft.TemplateEngine.Cli/CommandParsing/CommandParserSupport.cs

```diff
@@ -0,0 +1,222 @@
+                }
+            }
+
+            if (invalidParams.Count > 0)
+            {
+                string unusableDisplayList = string.Join(", ", invalidParams);
+                throw new Exception($"Template is malformed. The following parameter names are invalid: {unusableDisplayList}");
+            }
+
+            return GetNewCommandForTemplate(commandName, templateName, NewCommandVisibleArgs, NewCommandHiddenArgs, DebuggingCommandArgs, paramOptionList.ToArray());
+        }
+
+        private static Option[] NewCommandVisibleArgs
+        {
+            get
+            {
+                return new[]
+                {
+                    Create.Option("-l|--list", LocalizableStrings.ListsTemplates, Accept.NoArguments()),
+                    Create.Option("-lang|--language", LocalizableStrings.LanguageParameter,
+                                    // TODO: dynamically get the language set for all installed templates.
+                                    Accept.ExactlyOneArgument()
+                                        .WithSuggestionsFrom("C#", "F#")
+                                        .With(defaultValue: () => "C#")),
+                    Create.Option("-n|--name", LocalizableStrings.NameOfOutput, Accept.ExactlyOneArgument()),
+                    Create.Option("-o|--output", LocalizableStrings.OutputPath, Accept.ExactlyOneArgument()),
+                    Create.Option("-h|--help", LocalizableStrings.DisplaysHelp, Accept.NoArguments()),
+                    Create.Option("--type", LocalizableStrings.ShowsFilteredTemplates, Accept.ExactlyOneArgument()),
+                    Create.Option("--force", LocalizableStrings.ForcesTemplateCreation, Accept.NoArguments()),
+                };
+            }
+        }
+
+        private static Option[] NewCommandHiddenArgs
+        {
+            get
+            {
+                return new[]
+                {
+                    Create.Option("-a|--alias", string.Empty, Accept.ExactlyOneArgument()),
+                    Create.Option("-x|--extra-args", string.Empty, Accept.OneOrMoreArguments()),
+                    Create.Option("--locale", string.Empty, Accept.ExactlyOneArgument()),
+                    Create.Option("--quiet", string.Empty, Accept.NoArguments()),
+                    Create.Option("-i|--install", string.Empty, Accept.OneOrMoreArguments()),
+                    Create.Option("-all|--show-all", string.Empty, Accept.NoArguments()),
+                    Create.Option("--allow-scripts", string.Empty, Accept.ZeroOrOneArgument()),
+                };
+            }
+        }
+
+        private static Option[] NewCommandReservedArgs
+        {
+            get
+            {
+                return new[]
+                {
+                    Create.Option("-up|--update", string.Empty, Accept.ZeroOrOneArgument()),
+                    Create.Option("-u|--uninstall", string.Empty, Accept.ZeroOrOneArgument()),
```

Comment:
Ah, the issue is that I didn't make "install" & "uninstall" into generally visible args in this PR. I'll move them to their new (proper) place.